packages/medusa-plugin-algolia/src/types.ts

```diff
@@ -0,0 +1,18 @@
+import { SearchTypes } from "@medusajs/types"
+
+export type SearchOptions = {
+  paginationOptions: Record<string, unknown>
+  filter: string
+  additionalOptions: Record<string, unknown>
+}
+
+export type AlgoliaPluginOptions = {
+  applicationId: string
```

Comment:
suggestion: Reguarding the backward compat, depending on the choice regarding the previous comment about that, we should re add the previous prop as deprecated